### PR TITLE
Assume ports that look like Windows pipes are

### PR DESCRIPTION
**Pull request by zionts**
This matches what most APIs allow for in that you can listen on a pipe by passing it to the `port` variable. The only caveat for us is that we only support Windows named pipes at the moment, which are all in the special `\\.\pipe\` directory, so this match will only pass those through.

### CODE REVIEW

File: src/engine.ts [hunk 1]
```diff
@@ -224,9 +224,16 @@ export class ApolloEngine extends EventEmitter {
     if (options.port === undefined && options.pipePath === undefined) {
       throw new Error('Either `port` or `pipePath` must be defined');
     } else if (typeof options.port === 'string') {
-      options.port = parseInt(options.port, 10);
+      const stringPort = options.port;
+      options.port = parseInt(stringPort, 10);
       if (isNaN(options.port)) {
-        throw new Error(`port must be an integer, not '${options.port}'`);
+        // Engine only knows how to listen on Windows named pipes
+        if (stringPort.match(/^\\\\\.\\pipe\\/)) {
```
Comment (glasser) [inline src/engine.ts]: might be more clear using stringPort.startsWith... it at least lets you drop the backslash before the `.`.  (unfortunately you can't use `String.raw` here as I think you can't have a template string that ends with a backslash)

File: src/engine.ts [hunk 2]
```diff
@@ -224,9 +224,16 @@ export class ApolloEngine extends EventEmitter {
...
+          options.pipePath = stringPort;
+          options.port = undefined;
+        } else {
+          throw new Error(`port must be an integer, not '${options.port}'`);
```
Comment (glasser) [inline src/engine.ts]: error should say "or a Windows named pipe"

File: src/engine.ts (hunk 1 above)
Comment (zionts) [reply src/engine.ts]: I like using String.raw and just appending the escaped backslash.

File: src/engine.ts [hunk 3]
```diff
@@ -224,9 +224,18 @@ export class ApolloEngine extends EventEmitter {
...
+      options.port = parseInt(stringPort, 10);
       if (isNaN(options.port)) {
-        throw new Error(`port must be an integer, not '${options.port}'`);
+        // Engine only knows how to listen on Windows named pipes
+        const winPipePrefix = String.raw`\\.\pipe`;
+        if (stringPort.startsWith(winPipePrefix + '\\')) {
+          options.pipePath = stringPort;
+          options.port = undefined;
+        } else {
+          throw new Error(`port must be an integer or a Windows named pipe\
```
Comment (glasser) [inline src/engine.ts]: huh what's up with that wrapping?

Comment (zionts) [reply src/engine.ts]: avoiding the new line?

Comment (glasser) [reply src/engine.ts]: doesn't it leave in the spaces though?

**Pull request merged**